src/core/lib/props.js

```diff
@@ -122,35 +122,32 @@ function diffDataProps(props, oldProps) {
 // Checks if any update triggers have changed
 // also calls callback to invalidate attributes accordingly.
 function diffUpdateTriggers(props, oldProps, onUpdateTriggered) {
-  // const {attributeManager} = this.state;
-  // const updateTriggerMap = attributeManager.getUpdateTriggerMap();
   if (oldProps === null) {
     return 'oldProps is null, initial diff';
   }
 
-  let reason = false;
-
   // If the 'all' updateTrigger fires, ignore testing others
   if ('all' in props.updateTriggers) {
     const diffReason = diffUpdateTrigger(oldProps, props, 'all');
     if (diffReason) {
       onUpdateTriggered('all');
-      return diffReason;
+      return {all: true};
```

Comment:
If we do it like this, we can probably remove the clunky `onUpdateTriggered` callback, as the layer can just iterate over the keys it gets back.